src/sentry/sentry_apps/api/endpoints/sentry_app_requests_v2.py

```diff
@@ -0,0 +1,178 @@
+
+        requests: list[RpcSentryAppRequest] = []
+        region_filter: SentryAppRequestFilterArgs = {}
+        control_buffer = SentryAppWebhookRequestsBuffer(sentry_app)
+        control_errors_only = region_filter["errors_only"] = errors_only
+
+        def filter_and_append_requests(unfiltered_requests: list[RpcSentryAppRequest]) -> None:
+            for i, req in enumerate(unfiltered_requests):
+                if filter_by_date(req, start_time, end_time) and filter_by_organization(
+                    req, organization
+                ):
+                    requests.append(BufferedRequest(id=i, data=req))
+
+        def get_buffer_requests_for_control(event_type) -> list[RpcSentryAppRequest]:
+            control_requests = [
+                serialize_rpc_sentry_app_request(req)
+                for req in control_buffer.get_requests(
+                    event=event_type, errors_only=control_errors_only
+                )
+            ]
+            filter_and_append_requests(control_requests)
+
+        def get_buffer_requests_for_regions() -> list[RpcSentryAppRequest]:
+            for region_name in find_all_region_names():
+                buffer_requests = app_request_service.get_buffer_requests_for_region(
+                    sentry_app_id=sentry_app.id,
+                    region_name=region_name,
+                    filter=region_filter,
+                )
+                if buffer_requests is not None:
+                    filter_and_append_requests(buffer_requests)
+
+        # If event type is installation.created or installation.deleted, we only need to fetch requests from the control buffer
+        if event_type == "installation.created" or event_type == "installation.deleted":
+            get_buffer_requests_for_control(event_type)
+        # If event type has been specified, we only need to fetch requests from region buffers
+        elif event_type:
+            region_filter["event"] = event_type
+            get_buffer_requests_for_regions()
+        else:
+            control_event_type = [
+                "installation.created",
+                "installation.deleted",
+            ]
+            get_buffer_requests_for_control(control_event_type)
+            region_filter["event"] = list(
+                set(EXTENDED_VALID_EVENTS)
+                - {
+                    "installation.created",
+                    "installation.deleted",
+                }
+            )
+            get_buffer_requests_for_regions()
+
+        requests.sort(key=lambda x: parse_date(x.data.date), reverse=True)
+
+        def data_fn(offset, limit):
+            page_offset = offset * limit
+            return requests[page_offset : page_offset + limit]
+
+        return self.paginate(
+            request=request,
+            paginator=GenericOffsetPaginator(data_fn),
+            on_results=lambda x: serialize(x, request.user, RequestSerializer(sentry_app)),
+        )
```

Comment:
Will this paginate correctly? I'm concerned the results will be weird if an org has a high volume of webhooks and older request buffer responses are constantly rotating off.

Also, do we need to be paginating if our limit is ~100 buffers per region?

Reply:
Just updating here that we decided to remove pagination